deployment/fedora-package-x64/pkg-src/restart.sh

```diff
@@ -1,6 +1,36 @@
-#!/bin/sh
+#!/bin/bash
 
-NAME=jellyfin
-restart_cmd="/usr/bin/systemctl restart ${NAME}"
-echo "sleep 2; sudo $restart_cmd > /dev/null 2>&1" | at now > /dev/null 2>&1
-exit 0
\ No newline at end of file
+# restart.sh - Jellyfin server restart script
+# Part of the Jellyfin project (https://github.com/jellyfin)
+#
+# This script restarts the Jellyfin daemon on Linux when using
+# the Restart button on the admin dashboard. It supports the
+# systemctl, service, and traditional /etc/init.d (sysv) restart
+# methods, chosen automatically by which one is found first (in
+# that order).
+#
+# This script is used by the Debian/Ubuntu/Fedora/CentOS packages.
+
+get_service_command() {
+    for command in systemctl service; do
+        if which $command &>/dev/null; then
+            echo $command && return
+        fi
+    done
+    echo "sysv"
+}
+
+cmd="$( get_service_command )"
+echo "Using system control command '$cmd' to restart Jellyfin..."
+case $cmd in
+    'systemctl')
+        echo "sleep 2; /usr/bin/sudo /usr/bin/systemctl restart jellyfin" | at now 
+        ;;
+    'service')
+        echo "sleep 2; /usr/bin/sudo /sbin/service jellyfin restart" | at now 
+        ;;
+    'sysv')
+        echo "sleep 2; /usr/bin/sudo /etc/init.d/jellyfin restart" | at now 
```

Comment:
Why `at now` instead of executing the commands directly?

Reply:
Because how JF restarts is bass-ackwards. It terminates the process like it was shutting down, then at the last second calls this script. However by that point the parent context is exiting, and it terminates the children, including this script. So we have to use this gross hack (as Emby did) to schedule the "start up" job externally from the script - `at` just seems like the easiest way to do so.

Reply:
I hate it too, believe me, but without getting into core code changes to perform a proper restart without invoking the external service manager, we're stuck with this one :-1: